pkg/api/server/v1alpha2/records_test.go

```diff
@@ -557,3 +557,48 @@ func TestUpdateRecord(t *testing.T) {
 		})
 	}
 }
+
+func TestDeleteRecord(t *testing.T) {
```

Comment:
A related test that would be very useful is to test whether cascade deletes work as intended (i.e. if you delete the Result, the record is deleted as well).

This isn't directly testing the DeleteRecord RPC, so up to you if you want to include it in this PR or defer to another PR.

Reply:
I guess I can send out another PR and put the cascade delete test in the TestDeleteResult.